test/cri-containerd/policy_test.go

```diff
@@ -1259,3 +1261,75 @@ func Test_ExecInUVM_WithPolicy(t *testing.T) {
 		})
 	}
 }
+
+func Test_RunPodSandbox_Concurrently(t *testing.T) {
```

Comment:
Can [`t.Parallel`](https://pkg.go.dev/testing#T.Parallel) be used here instead of errgroups, so we can keep and the `testing` helper functions?

Reply:
Can you elaborate on how this would work? Documentation suggests that parallel tests can be run only with other parallel tests and that the same test won't be run in parallel with itself when `test.count` is used?

Reply:
`test.count` runs the entire test repeated times. from what I've seen, its mostly used with benchmarking:

```go
func TestCount(t *testing.T) {
	time.Sleep(50 * time.Millisecond)
	t.Log("hi")
}
```

```powershell 
> go test -v -count 2 
=== RUN   TestCount
    p_test.go:10: hi
--- PASS: TestCount (0.07s)
=== RUN   TestCount
    p_test.go:10: hi
--- PASS: TestCount (0.06s)
PASS
```
For `t.Parallel`, calling it from sub-tests would be exactly what you want, and function similar to launching go-routines or `eg.Go()`:

```go
func Test_Parallel(t *testing.T) {
	for i := 0; i < 3; i++ {
		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
			t.Parallel()
			t.Log(t.Name(), "start")
			time.Sleep(50 * time.Millisecond)
			t.Log(t.Name(), "done")
		})
	}


	t.Run("not", func(t *testing.T) {
		t.Log("im not parallel :(")
	})
}
```

yields:

```powershell
> go test -v
=== RUN   Test_Parallel
=== RUN   Test_Parallel/0
=== PAUSE Test_Parallel/0
=== RUN   Test_Parallel/1
=== PAUSE Test_Parallel/1
=== RUN   Test_Parallel/2
=== PAUSE Test_Parallel/2
=== RUN   Test_Parallel/not
    p_test.go:20: im not parallel :(
=== CONT  Test_Parallel/0
    p_test.go:13: Test_Parallel/0 start
=== CONT  Test_Parallel/1
    p_test.go:13: Test_Parallel/1 start
=== CONT  Test_Parallel/2
    p_test.go:13: Test_Parallel/2 start
    p_test.go:15: Test_Parallel/2 done
=== CONT  Test_Parallel/0
    p_test.go:15: Test_Parallel/0 done
=== CONT  Test_Parallel/1
    p_test.go:15: Test_Parallel/1 done
--- PASS: Test_Parallel (0.00s)
    --- PASS: Test_Parallel/not (0.00s)
    --- PASS: Test_Parallel/2 (0.05s)
    --- PASS: Test_Parallel/0 (0.05s)
    --- PASS: Test_Parallel/1 (0.05s)
PASS
```
from what I can tell, calling `t.Parallel` yeilds to the parent test, which then can start other sub-tests, and then the parent test starts all the parallel sub tests when it reaches the end of its execution